ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -338,4 +370,56 @@ private void addStatTask(ASTNode root, Table table, Path oldPartitionLocation, P
       moveTask.addDependentTask(statTask);
     }
   }
+
+  public StringBuilder constructDeleteQuery(Table table, Map<String, String> partitionSpec) throws SemanticException {
+    String qualifiedTableName = HiveTableName.ofNullable(HiveUtils.unparseIdentifier(table.getTableName(), this.conf),
```

Comment:
can we reuse the getFullTableNameForSQL to get the qualifiedTableName?

Reply:
`getFullTableNameForSQL ` is used for ASTNode.

Reply:
ok, so? pass the the evaluated name from root method

Reply:
Done.